gubernator/github_auth_test.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import urlparse
+
+import webtest
+
+import gcs_async_test
+import main
+
+main.app.config['github_client'] = {
+    'id': '12345',
+    'secret': 'swordfish',
+}
+
+app = webtest.TestApp(main.app)
+
+VEND_URL = 'https://github.com/login/oauth/access_token'
+USER_URL = 'https://api.github.com/user'
+
+class TestGithubAuth(unittest.TestCase):
```

Comment:
The phase1, phase2 functions in here are pretty slick and make for some easy to understand test cases!